shared/wallets/trustline/index.tsx

```diff
@@ -3,100 +3,178 @@ import * as React from 'react'
+  searchingAssets?: Array<Types.AssetID>
+  totalAssetsCount?: number
+  waitingSearch: boolean
 }
 
-type Props = BodyProps & {
+type BodyProps = _Props & {onFocusChange?: (focused: boolean) => void}
+
+type Props = _Props & {
   onDone: () => void
 }
 
 const makeSections = (props: BodyProps) => [
   ...(props.searchingAssets
     ? [
         {
-          data: props.searchingAssets.toArray(),
+          data: props.searchingAssets,
           key: 'section-search',
-          keyExtractor: item => item,
+          keyExtractor: item => `search-item:${item}`,
           title: '',
         },
       ]
     : []),
-  ...(!props.searchingAssets && props.acceptedAssets.size
+  ...(!props.searchingAssets && props.acceptedAssets.length
     ? [
         {
-          data: props.acceptedAssets.toArray(),
+          data: props.acceptedAssets,
           key: 'section-accepted',
-          keyExtractor: item => item,
+          keyExtractor: item => `accepted-item:${item}`,
           title: 'Accepted assets',
         },
       ]
     : []),
-  ...(!props.searchingAssets && props.popularAssets.size
+  ...(!props.searchingAssets && props.popularAssets.length
     ? [
         {
-          data: props.popularAssets.toArray(),
+          data: props.popularAssets,
           key: 'section-popular',
-          keyExtractor: item => item,
+          keyExtractor: item => `popular-item:${item}`,
           title: 'Popular assets',
         },
       ]
     : []),
 ]
 
+// hack around the bug where when we change from search mode where there's no
+// section header, into normal mode where there are section headers, first
+// section header doesn't show.
+const getSectionListKey = (props: BodyProps) =>
+  `sl-${props.searchingAssets ? 'sa' : '_'}-${props.acceptedAssets.length ? 'aa' : '_'}-${
+    props.popularAssets.length ? 'pa' : '_'
+  }`
+
 const sectionHeader = section =>
   !section.title || (
     <Kb.Box2 direction="horizontal" alignItems="center" fullWidth={true} style={styles.sectionHeader}>
       <Kb.Text type="BodySmall">{section.title}</Kb.Text>
     </Kb.Box2>
   )
 
-const Body = (props: BodyProps) => (
-  <Kb.Box2 direction="vertical" fullWidth={true} style={styles.body}>
-    <Kb.Box2 direction="horizontal" fullWidth={true} style={styles.searchFilter}>
-      <Kb.SearchFilter
-        icon="iconfont-search"
-        fullWidth={true}
-        placeholderText={`Search ${props.totalAssetsCount} assets`}
-        onChange={props.onSearchChange}
-      />
-    </Kb.Box2>
-    <Kb.Divider />
-    {!!props.errorMessage && <Kb.Banner color="red" text={props.errorMessage} />}
+const haveEnoughBalanceToAccept = memoize(
+  (props: BodyProps) => props.balanceAvailableToSend >= Constants.trustlineHoldingBalance
+)
+
+const ListUpdateOnMount = (props: BodyProps) => {
+  // hack to get `ReactList` to render more than one item on initial mount.
+  // Somehow we need two updates here.
+  const [updateCounter, setUpdateCounter] = React.useState(0)
```

Comment:
please make a ticket to remove this later. i don't think this is necessary.

Reply:
make HOTPOT-197